models/zeroex/optimism/zeroex_api_optimism_fills.sql

```diff
@@ -0,0 +1,394 @@
+            plp.evt_index               AS evt_index,
+            plp.contract_address,
+            plp.evt_block_time          AS block_time,
+            provider                    AS maker,
+            recipient                   AS taker,
+            inputToken                  AS taker_token,
+            outputToken                 AS maker_token,
+            inputTokenAmount            AS taker_token_amount_raw,
+            outputTokenAmount           AS maker_token_amount_raw,
+            'LiquidityProviderSwap'     AS type,
+            zeroex_tx.affiliate_address AS affiliate_address,
+            TRUE                        AS swap_flag,
+            FALSE                       AS matcha_limit_order_flag
+    FROM {{ source('zeroex_optimism', 'ExchangeProxy_evt_LiquidityProviderSwap') }} plp
+    JOIN zeroex_tx ON zeroex_tx.tx_hash = plp.evt_tx_hash
+
+    {% if is_incremental() %}
+    WHERE evt_block_time >= date_trunc('day', now() - interval '1 week')
+    {% endif %}
+    {% if not is_incremental() %}
+    WHERE evt_block_time >= '{{zeroex_v3_start_date}}'
+    {% endif %}
+),
+direct_uniswapv3 AS (
+    SELECT 
+            swap.evt_tx_hash                                                                        AS tx_hash,
+            swap.evt_index,
+            swap.contract_address,
+            swap.evt_block_time                                                                     AS block_time,
+            swap.contract_address                                                                   AS maker,
+            LAST_VALUE(swap.recipient) OVER (PARTITION BY swap.evt_tx_hash ORDER BY swap.evt_index) AS taker,
+            CASE WHEN amount0 < '0' THEN pair.token1 ELSE pair.token0 END                           AS taker_token,
+            CASE WHEN amount0 < '0' THEN pair.token0 ELSE pair.token1 END                           AS maker_token,
+            CASE
+                WHEN amount0 < '0' THEN abs(swap.amount1)
+                ELSE abs(swap.amount0) END                                                          AS taker_token_amount_raw,
+            CASE
+                WHEN amount0 < '0' THEN abs(swap.amount0)
+                ELSE abs(swap.amount1) END                                                          AS maker_token_amount_raw,
+            'Uniswap V3 Direct'                                                                     AS type,
+            zeroex_tx.affiliate_address                                                             AS affiliate_address,
+            TRUE                                                                                    AS swap_flag,
+            FALSE                                                                                   AS matcha_limit_order_flag
+    FROM {{ source('uniswap_v3_optimism', 'Pair_evt_Swap') }} swap
+   LEFT JOIN {{ source('uniswap_v3_optimism', 'Factory_evt_PoolCreated') }} pair ON pair.pool = swap.contract_address
```

Comment:
Hi @RantumBits, you may need to add/define this source in the source.yml file as currently [the test failure](https://github.com/duneanalytics/spellbook/actions/runs/3597575181/jobs/6059457094#step:7:11) is because of this issue.

Moreover the `uniswap_v3_optimism.Factory_evt_PoolCreated` is undefined but `uniswap_v3_optimism.factory_evt_poolcreated` is defined in the uniswap_op_source.yml. These two may be the same table. The source name is case sensitive.

Reply:
Thanks @Hosuke I changed case and resolved that issue